gtda/homology/simplicial.py

```diff
@@ -721,3 +722,272 @@ def plot(Xt, sample=0, homology_dimensions=None):
         """
         return plot_diagram(
             Xt[sample], homology_dimensions=homology_dimensions)
+
+
+@adapt_fit_transform_docs
+class FlagserPersistence(BaseEstimator, TransformerMixin, PlotterMixin):
+    """:ref:`Persistence diagrams <persistence_diagram>` resulting from
+    :ref:`filtrations <filtered_complex>` of :ref:`directed or undirected flag
+    complexes <clique_and_flag_complexes>`.
+
+    Given a weighted directed or undirected graph, information about the
+    appearance and disappearance of topological features (technically,
+    :ref:`homology classes <homology_and_cohomology>`) of various dimension
+    and at different scales is summarised in the corresponding persistence
+    diagram.
+
+    Parameters
+    ----------
+    homology_dimensions : list or tuple, optional, default: ``(0, 1)``
+        Dimensions (non-negative integers) of the topological features to be
+        detected.
+
+    directed : bool, optional, default: ``True``
+        If ``True``, :meth:`transform` computes the persistence diagrams of
+        filtered directed flag complexes arising from weighted directed
+        graphs. If False, computes the persistence diagrams of undirected
+        filtered flag complexes obtained by considering all weighted edges of
+        each weighted undirected graphs of the input collection as undirected,
+        and it is therefore sufficient (but not necessary) to pass a collection
+        of upper-triangular matrices. When ``False``, in each weighted
+        undirected graphs of the input collection, if two directed edges
+        corresponding to the same undirected edge are assigned different
+        weights, only the one on the upper triangular part of the adjacency
+        matrix is considered.
+
+    filtration : string, optional, default: ``'max'``
+        Algorithm determining the filtration. Warning: if an edge filtration is
```

Comment:
What does this warning really mean to the user? Presumably the "possible values" below all make "consistent filtrations", and if the user cannot pass custom algorithms then why should we tell them this?

Reply:
I believe this statement should be replaced by : "It is assumed that a vertex weight cannot be larger than the weight of the edges it forms."

Reply:
But then this doesn't have anything to do with the filtration algorithm and all to do with the input right?

Reply:
Wait, no, I was wrong. I do believe that this was for the custom filtration that C++ `flagser` allows.

My comment above does apply to the input indeed.

Reply:
Ok, so how about adding your sentence above to the docs for `X` in `fit` and `transform`? And removing this warning entirely from here?

Reply:
Done, thanks!